pylint/lint/run.py

```diff
@@ -190,6 +190,11 @@ def __init__(
                 linter.set_option("jobs", 1)
             elif linter.config.jobs == 0:
                 linter.config.jobs = _cpu_count()
+                if sys.platform == "win32":
```

Comment:
Doesn't it make more sense to de-dent this one level and always running this? Perhaps even with a nice error message to `stderr` saying that we have limited the jobs count from ... to 56?

Reply:
It's a judgment call. The wpt project felt it made more sense to not override explicit settings, but doing so with a warning message also makes perfect sense. I don't have strong feelings either way.

Reply:
Hm, yeah perhaps that's indeed better. If we overwrite by default there would be no way for a user to overwrite it themselves.

But then I think it should be in the `_cpu_count()` function, as that is responsible for getting the "correct" number so this fix belongs to its responsibilities.

Could you also add a short note about this here:
https://github.com/PyCQA/pylint/blob/17c8f26f35fbf05d149da9719bab17a148f2e363/pylint/lint/base_options.py#L247

That will automatically make it show up in our settings documentation!

Reply:
Done.